src/rvya25-profile.adoc

```diff
@@ -0,0 +1,466 @@
+
+The following unprivileged extensions are mandatory:
+
+- The RVYA25S64 mandatory unprivileged extensions include all the
+mandatory unprivileged extensions in RVYA25U64.
+
+- *Zifencei*  Instruction-Fetch Fence.
+
+NOTE: Zifencei is mandated as it is the only standard way to support
+instruction-cache coherence in RVYA25 application processors.  A new
+instruction-cache coherence mechanism is under development
+(tentatively named Zjid) which might be added as an option in the
+future.
+
+The following privileged extensions are mandatory:
+
+- *Ss1p13*  Supervisor architecture version 1.13.
+
+- *Svbare* The `satp` mode Bare must be supported.
+
+- *Sv39* Page-based 39-bit virtual-Memory system.
+
+- *Svade* Page-fault exceptions are raised when a page is accessed
+   when A bit is clear, or written when D bit is clear.
+
+- *Ssccptr* Main memory regions with both the cacheability and
+   coherence PMAs must support hardware page-table reads.
+
+- *Sstvecd* `stvec.MODE` must be capable of holding the value 0
+  (Direct).  When `stvec.MODE=Direct`, `stvec.BASE` must be capable of
+  holding any valid four-byte-aligned address.
+
+- *Sstvala* `stval` must be written with the faulting virtual address
+  for load, store, and instruction page-fault, access-fault, and
+  misaligned exceptions, and for breakpoint exceptions other than
+  those caused by execution of the `EBREAK` or `C.EBREAK` instructions.
+  For virtual-instruction and illegal-instruction exceptions, `stval` must be written with the
+  faulting instruction.
+
+- *Sscounterenw* For any `hpmcounter` that is not read-only zero, the
+  corresponding bit in `scounteren` must be writable.
+
+- *Svpbmt* Page-based memory types
+
+- *Svinval* Fine-grained address-translation cache invalidation.
+
+- *Svnapot* NAPOT translation contiguity.
+
+- *Sstc* supervisor-mode timer interrupts.
+
+- *Sscofpmf* count overflow and mode-based filtering.
+
+- *Ssu64xl* `sstatus.UXL` must be capable of holding the value 2
+(i.e., UXLEN=64 must be supported).
+
+- *Sha* The augmented hypervisor extension.
+
+NOTE: Sha is defined in the RVA23 profile document.
+
+The following mandatory extension is RVY25S64 specific:
+
+- *Svy* Read/write access control for capability tags in the page table entries.
+
+The following extension is mandatory in RVA23 but is _incompatible_ with RVY:
+
+- *Ssnpm* Pointer masking, with `senvcfg.PMM` and `henvcfg.PMM` supporting,
+at minimum, settings PMLEN=0 and PMLEN=7.
+
+NOTE: How to benefit from the combination of pointer masking and CHERI is a research topic.
+
+==== RVYA25S64 Optional Extensions
+
+===== Localized Options
+
+There are no privileged localized options in RVYA25S64.
+
+===== Development Options
+
+There privileged development options in RVYA25S64.
+
+- *Svucrg* Page-based revocation (preventing use-after-free in virtual memory systems).
+
+- *Svucrglct* (implies *Svucrg*) checking loaded capability tags before taking exceptions to optimize revocation.
```

Comment:
I remain unconvinced that having this be optional is a good idea, but that's a separate issue for the RVY spec itself about whether this extension should even exist or just be part of Svucrg

Reply:
we had a useful discussion about it today, @nwf pionted out that if you support synchronous ECC errors on loaded data then you should support the tag checking version, and if not, then you supoprt the simpler version which doesn't check tags.